aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsCodegenDecorator.kt

```diff
@@ -43,7 +43,9 @@ val DECORATORS: List<ClientCodegenDecorator> =
             UserAgentDecorator(),
             SigV4AuthDecorator(),
             HttpRequestChecksumDecorator(),
+            HttpRequestChecksumMutationInterceptorGenerator(),
             HttpResponseChecksumDecorator(),
+            HttpResponseChecksumMutationInterceptorGenerator(),
```

Comment:
Should they be called `***Decorator` rather than `***Generator` since they inherit from `ClientCodegenDecorator`?

Reply:
Good catch, those names predated me finding the `OperationSection.RuntimePluginSupportingTypes` where these now live. Originally the codegen for these was based on the `EndpointParamsInterceptorGenerator` and I adopted the name from there.

Reply:
Updated in https://github.com/smithy-lang/smithy-rs/pull/3845/commits/0c6059b5a5a5aed884fe927b22c3670ef3cd57d6